aspnetcore/web-api/action-return-types.md

```diff
@@ -10,6 +10,176 @@ uid: web-api/action-return-types
+
+C# doesn't support implicit cast operators on interfaces. Consequently, conversion of the interface to a concrete type is necessary to use `ActionResult<T>`. For example, use of `IEnumerable` in the following example doesn't work:
+
+```csharp
+[HttpGet]
+public ActionResult<IEnumerable<Product>> Get() =>
+    _repository.GetProducts();
+```
+
+One option to fix the preceding code is to return `_repository.GetProducts().ToList();`.
+
+Most actions have a specific return type. Unexpected conditions can occur during action execution, in which case the specific type isn't returned. For example, an action's input parameter may fail model validation. In such a case, it's common to return the appropriate `ActionResult` type instead of the specific type.
+
+### Synchronous action
+
+Consider a synchronous action in which there are two possible return types:
+
+:::code language="csharp" source="~/../AspNetCore.Docs.Samples/mvc/action-return-types/7.x/WebApiSample/Controllers/ActionResultOfTProductsController.cs" id="snippet_GetByIdActionResultOfT" highlight="7":::
+
+In the preceding action:
+
+* A 404 status code is returned when the product doesn't exist in the database.
+* A 200 status code is returned with the corresponding `Product` object when the product does exist.
+
+### Asynchronous action
+
+Consider an asynchronous action in which there are two possible return types:
+
+:::code language="csharp" source="~/../AspNetCore.Docs.Samples/mvc/action-return-types/7.x/WebApiSample/Controllers/ActionResultOfTProductsController.cs" id="snippet_CreateAsyncActionResultOfT" highlight="9,15":::
+
+In the preceding action:
+
+* A 400 status code (<xref:Microsoft.AspNetCore.Mvc.ControllerBase.BadRequest%2A>) is returned by the ASP.NET Core runtime when:
+  * The [`[ApiController]`](xref:Microsoft.AspNetCore.Mvc.ApiControllerAttribute) attribute has been applied and model validation fails.
+  * The product description contains "XYZ Widget".
+* A 201 status code is generated by the <xref:Microsoft.AspNetCore.Mvc.ControllerBase.CreatedAtAction%2A> method when a product is created. In this code path, the `Product` object is provided in the response body. A `Location` response header containing the newly created product's URL is provided.
+
+## HttpResults type
+
+TODO
+
+<!-- Mention no content-negotiation -->
+
+### IResult
+
+:::code language="csharp" source="~/../AspNetCore.Docs.Samples/mvc/action-return-types/7.x/WebApiSample/Controllers/IResultProductsController.cs" id="snippet_GetByIdIResult" highlight="7":::
+
+In the preceding action:
+
+* A 404 status code is returned when the product doesn't exist in the database.
+* A 200 status code is returned with the corresponding `Product` object when the product does exist, generated by the [Results.Ok\<T>()](<xref:Microsoft.AspNetCore.Http.Results.Ok>).
```

Comment:
@guardrex  your tool has `<xref:Microsoft.AspNetCore.Http.Results.Ok(Object)>`

Reply:
It doesn't work very well any longer. I think that there were changes on the API end that I never reacted to. I don't currently have plans to work on it further. Is the one that Kirk made still running? I had the impression that his version was providing better (*correct!*) links.

Reply:
I think it's https://github.com/serpent5/Serpent5.Xrefs 
I had trouble deploying it. @guardrex  maybe you could help me get it deployed.
cc @serpent5

Reply:
![mccoy](https://user-images.githubusercontent.com/1622880/193427376-0f5dc3a3-7dae-4cfb-8d74-a1b2d699a927.png)

### *Damn it Rick! I'm a .NET guy, not a Node JSX dev!* 🤣

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&mdash; Dr. Leonard H. "Bones" McCoy ([DeForest Kelley](https://en.wikipedia.org/wiki/DeForest_Kelley)) [Star Trek](http://www.startrek.com/), ©1966-69 [Paramount Pictures](http://www.paramount.com)

Reply:
Yeah, that's the repo. It was deployed as part of my MSFT subscription, but that was deleted. 

@Rick-Anderson I'll email you tomorrow with some instructions on how to deploy it.

Reply:
Thx @serpent5 ... That *RexHackCrap*&trade; that I built has broken down. If Rick gets your app running, I'll start using that. I don't have the time or interest to fix the one that I built.

Reply:
> Yeah, that's the repo. It was deployed as part of my MSFT subscription, but that was deleted.
> 
Might be easier to get you MVP status again :)